Meta kernel for _pack_padded_sequence

angelayi opened the pull request:
Summary: Op implementation: https://github.com/pytorch/pytorch/blob/8cf54929e3c553dca57146c0b2c2aecb596d8e65/aten/src/ATen/native/PackedSequence.cpp#L34

Fixes https://fb.workplace.com/groups/pytorch.edge.users/permalink/1499571650913123/

I'm not entirely sure how to test this meta kernel.

Differential Revision: D56478332

ezyang commented on torch/_subclasses/fake_impls.py:
```diff
@@ -922,6 +922,31 @@ def convert_tensor(t, device):
+        inputs = inputs.transpose(0, 1)
+
+    batch_size = inputs.size(1)
+    res_size = (batch_size, *inputs.shape[2:])
```
Isn't this just inputs.shape[1:]?

angelayi replied:
oh you're right 😅 thanks!

ezyang reviewed (approved):
Thanks!

angelayi committed:
Meta kernel for _pack_padded_sequence (#124794)

Summary:

Op implementation: https://github.com/pytorch/pytorch/blob/8cf54929e3c553dca57146c0b2c2aecb596d8e65/aten/src/ATen/native/PackedSequence.cpp#L34

Test Plan: CI

Differential Revision: D56478332

angelayi commented:
@pytorchbot merge

angelayi commented:
@pytorchbot merge
